addons/project/tests/test_access_rights.py

```diff
@@ -412,3 +412,11 @@ def test_project_user_cannot_read_private_task_of_another_user(self):
     def test_project_user_cannot_unlink_private_task_of_another_user(self):
         with self.assertRaises(AccessError):
             self.private_task.with_user(self.env.user).unlink()
+
+    def test_of_setting_root_user_on_private_task(self):
+        test_task = self.env['project.task'].create({
+            'name':'Test Private Task',
+            'project_id': False,
```

Comment:
You can just not include the field in the values, and it will be False by default